src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx

```diff
@@ -137,15 +137,19 @@ export default class TaskRoleContainerList extends React.Component {
       monacoProps: null,
       monacoTitle: '',
       monacoFooterButton: null,
-      logUrl: null,
+      fullLogUrls: null,
+      tailLogUrls: null,
+      tailLogUrlsCvt: null,
```

Comment:
Why there is `tailLogUrls` and `tailLogUrlsCvt`?

Reply:
Because I can't get url by `tailLogUrls[logType]`, and I think the easy way to get this url is `this.convertObjectFormat(tailLogUrls)[logType]`. This code appears 5 times, so I put it in the state to reduce calling function `convertObjectFormat()`. 

Or maybe we can change logic from "check a specific url" to "check all urls", that is, from `prevState.tailLogUrls[logType] === tailLogUrls[logType]` to `prevState.tailLogUrls === tailLogUrls`.(But maybe this will cause other problem?)

Reply:
Already fixed